### ISSUE TITLE
Testing fails

### ISSUE THREAD

**Issue by KuechlerO** [Oshlack/Slinker#3]
I think there are some Java-dependencies involved, which are not documented, but are necessary to execute the test run.

I get the following error:
```
warning: usleep is deprecated, and will be removed in near future!
warning: use "sleep 0.1" instead...
./tools/bpipe-0.9.9.9/bin/bpipe: Zeile 778: kill: (494943) - Kein passender Prozess gefunden
bash: Zeile 2: exec: java: Nicht gefunden.
```

**Comment by KuechlerO** [comment]
Suggested solution in your Slinker environment: 
`conda install openjdk`

**Comment by KuechlerO** [comment]
I've spend a lot of time now on trying to make this run, but I just cannot succeed... 

For me there are a lot of things going wrong with the packaging structure. I hacked a lot around in my efforts trying to fix that for me, but even though I made progress, I'm stuck at the last step (Visualization)...

Could you please verify, that the package structure is correct? And do you have further recommendations on how to actually execute the scripts? (The tutorial has only sparsely information available for that).

Here's my output:
```
=================================== Stage star_index (unhealthy) ===================================

====================================== Stage visualise (all) =======================================
Traceback (most recent call last):
  File "/sc-projects/sc-proj-btg/olik_splicing_project/Slinker/scripts/slinker_vis.py", line 16, in <module>
    from tools.Slinker.Slinker import slinker as sl
ModuleNotFoundError: No module named 'tools'
ERROR: stage visualise failed: Command in stage visualise failed with exit status = 1 : 

python /sc-projects/sc-proj-btg/olik_splicing_project/Slinker/scripts/slinker_vis.py ETV6 ./tests/unhealthy_Aligned.sortedByCoord.out.bam */%_Aligned.sortedByCoord.out.bam ./test_output//ETV6/resources /sc-projects/sc-proj-btg/olik_splicing_project/Slinker/test_output/ETV6/plots 1000 10 False 

```

**Comment by breons** [comment]
Hi there,

Sorry this has been a trial to get working. I'll add in the java dependency in the documentation.

Just checking that you have installed and activated the conda environment?

Also try checking whether you have these installed: https://github.com/Oshlack/Slinker/wiki/1b.-Manual-Installation#install-internal-dependencies

Did you use the automated installation script or the manual? 

Cheers,
Breon

**Comment by KuechlerO** [comment]
Mhm yeah, weird. My first installation was done on a cluster and caused troubles, but I retried on my local machine and there I was able to execute everything.

Just some small description of some of the hurdles I faced (maybe it helps other users):
1. First I followed the semi-automated installation (including openjdk)
2. Then I executed the `./dependencies_linux64.sh download` script: Here not all tools were successfully noted in the `workflows\tools.groovy`-file, but for me it was fine to just add the pathways to the tools, that are installed in the `tools` folder manually
3. Setting command line variables
4. Then I tried to execute the test -> But this failed obviously, due to the not configured `workflows/bpipe.config`-file. I think this confused me also, since I never got in touch with bpipe before and the tutorial did not mention much about it.
5. So I went ahead, deleted the bpipe2.config-file and retried.
6. This time there was an error: 
```========================================= Stage create_st ==========================================
Error: file/directory /Users/oliverkuchler/Studium/Master_BioInfo_Berlin/Masterthesis/Topic_Splicing_Variants/github_forks/Slinker/references/Homo_sapiens.GRCh37.dna.primary_assembly.fa does not exist!
ERROR: stage create_st failed: Command in stage create_st failed with exit status = 1 : 
```
7. This was fixed by renaming the `Homo_sapiens.GRCh37.75.dna.primary_assembly.fa` into `Homo_sapiens.GRCh37.dna.primary_assembly.fa`
8. Rerun & "Yeah! Results are produced!"


Thx very much for your work. I will now go on and further explore and see how to apply it to my data :)
